woudc_data_registry/models.py

```diff
@@ -1163,12 +1250,113 @@ def sync(ctx):
         Station,
         Instrument,
         Deployment,
-        DataRecord
+        DataRecord,
     ]
+    contribution_models = list()
+
+    """function to form all the contributions from the other models lists"""
+    def build_contributions(contribution_models, project_models,
+                            station_models, instrument_models, country_models,
+                            deployment_models, contributor_models):
```

Comment:
This function is indented inside the body of the enclosing function. While putting functions inside other functions is generally bad, it's especially true for long functions like this one.

Somewhere in the file is a space for loose functions like `unpack_station_names`, this should go around there.
Also, are all those parameters really necessary?